src/content/basics/cluster-size-autoscaling/index.md

```diff
@@ -0,0 +1,60 @@
++++
+title = "Cluster Size and Autoscaling"
+description = "This article explains options you have for defining the size of a Kubernetes cluster with Giant Swarm, and automatically scaling it"
+date = "2019-01-17"
+weight = 45
+type = "page"
+categories = ["basics"]
++++
+
+# Cluster Size and Autoscaling
+
+Starting with release version 6.3.0 for AWS, you can leverage the benefits of the [Kubernetes autoscaler](https://github.com/kubernetes/autoscaler) to define the number of worker nodes in a cluster based on demand.
+The autoscaler is provided with every cluster of version >= 6.3.0 on AWS.
+
+On Giant Swarm installations on Azure, on bare-metal, and on AWS prior to version 6.3.0, the cluster size would be defined statically.
+
+## Setting scaling limits
+
+With the autoscaler taking control over the number of worker nodes, you only define the bounds or limits in which the cluster size can vary. This is possible both during cluster creation as well as after creation.
+
+To enforce an exact cluster size and **effectively disable the autoscaler**, simply set the minimum and maximum worker node count to the same value. This is also the recommended way in case you want to control the number of worker nodes through some external tooling via the Giant Swarm API.
+
+## Minimal and default cluster size
+
+When creating a cluster without specifying the number of worker nodes, **three** worker nodes will be created. This is also the minimal number supported by Giant Swarm. On AWS starting with release version 6.3.0, when not specified, the maximum number of worker nodes is also set to **three**.
+
+Technically, while you may be able to create and run smaller clusters successfully, we don't encourage this due to reduced resilience. We explicitly deactivate all sorts of alerts for clusters with less than three worker nodes and won't get notified in case of any problems.
+
+## Minimal worker node instance type
+
+Relying on autoscaling may have an effect on the instance type you should use for your cluster.
+To explain this more detail, it is helpful to explain how the autoscaler determines whether a node is needed or not.
+
+The autoscaler periodically calculates the utilization of a node based on CPU and memory requests, compared to the node's total capacity. The utilization is compared to a configurable _utilization threshold_, which is 50% by default. If the utilization is below the threshold, the autoscaler decides to remove the node. If it's above, it will keep the node.
```

Comment:
Maybe I would somehow empathize that this is about kubernetes requests that are set on pods, otherwise, people might assume it's about the real utilization on the node which is not true. This then explains a little bit more than the issue with utilization over 50% as the critical components have set request to certain value but don't really have much real usage on an empty cluster.

Reply:
In other words: Utilization that `cluster-autoscaler` considers is the same as what k8s scheduler is working with. It's scheduling hints provided as `requests` settings in PODs and it's not directly connected to real CPU/memory utilization. When these scheduling hints don't represent the real workload that corresponding POD is producing, it results either in underutilization of HW resources (i.e. CPU is sitting idle or gigabytes of memory unused) or overloading of a node where the whole node goes unavailable due to load.

Reply:
Get it. Will add some info. This means we'll repeat ourselves compared to other articles, but that's fine.

Reply:
also this PR https://github.com/giantswarm/kubernetes-cluster-autoscaler/pull/14 seems to change the value to 65 %

Reply:
Please don't fix this to 65% or something yet. I will sync with marian when we have a definitive final value.

Reply:
This is now encoded in one place a so-called "shortcode" which is then re-used wherever necessary in docs.